aep/general/0134/aep.md.j2

```diff
@@ -0,0 +1,321 @@
+# Update
+
+In REST APIs, it is customary to make a `PATCH` or `PUT` request to a
+resource's URI (for example, `/v1/publishers/{publisher}/books/{book}`) in
+order to update that resource.
+
+Resource-oriented design (AEP-121) honors this pattern through the `Update`
+method (which mirrors the REST `PATCH` behavior). These RPCs accept the URI
+representing that resource and return the resource.
+
+## Guidance
+
+APIs **should** provide an update method for resources unless it is not
+valuable for users to do so. The purpose of the update method is to make
+changes to the resources without causing side effects.
+
+Update methods are specified using the following pattern:
+
+- The RPC's name **must** begin with the word `Update`. The remainder of the
+  RPC name **should** be the singular form of the resource's message name.
+- The request message **must** exactly match the RPC name, with a `Request`
+  suffix.
+- The response message **must** be the resource itself.
+  - The response **should** include the fully-populated resource, and **must**
+    include any fields that were sent and included in the update mask unless
+    they are input only (see AEP-203).
+  - If the update RPC is [long-running](#long-running-update), the response
+    message **must** be an `Operation` for which the return type is the
+    resource itself.
+- The method **should** support partial resource update, and the HTTP verb
+  **should** be `PATCH`.
+
+{% tab proto %}
+
+```proto
+rpc UpdateBook(UpdateBookRequest) returns (Book) {
+  option (google.api.http) = {
+    patch: "/v1/{book.path=publishers/*/books/*}"
+    body: "book"
+  };
+  option (google.api.method_signature) = "book,update_mask";
+}
+```
+
+- The resource's `path` field **should** map to the URI path.
+  - The `{resource}.path` field **should** be the only variable in the URI
+    path.
+- There **must** be a `body` key in the `google.api.http` annotation, and it
+  **must** map to the resource field in the request message.
+  - All remaining fields **should** map to URI query parameters.
+- There **should** be exactly one `google.api.method_signature` annotation,
+  with a value of `"{resource},update_mask"`.
+
+**Note:** Unlike the other four standard methods, the URI path here references
```

Comment:
this is one of the things I like the least about this AEP - it's the only one that doesn't have path in the request.

I think this is something that, for usability's sake, should deviate from the AEPs.

Reply:
Do you mean it should deviate from the AIPs?  I'm not sure exactly what change you're recommending.

Update needs the resource in the request, which includes the path field; having a separate path field is redundant, no?

(Also, create also doesn't have path in the request.)

Reply:
Created https://github.com/aep-dev/aep.dev/issues/123

Reply:
create by necessity can't have a path in the request, at least not a required one: that would deviate in the case of server-generated fields.

In general apply is a simpler for that reason.

My idea would be to have path in the request, and the path in the resource always be output-only.

But I'll file a ticket on this.